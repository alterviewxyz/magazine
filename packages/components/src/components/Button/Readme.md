Button example:

```js
<Button>Hey There</Button>
```
