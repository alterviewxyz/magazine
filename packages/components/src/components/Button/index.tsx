import React, { FC, MouseEvent } from "react";
import classNames from "classnames";

interface IButton {
  disabled?: boolean;
  onClick?: (event: MouseEvent<HTMLButtonElement>) => void;
  className?: string;
}

const Button: FC<IButton> = ({
  children,
  disabled = false,
  onClick,
  className,
}) => {
  const buttonClasses = classNames({
    btn: true,
  });

  return (
    <button
      disabled={disabled}
      onClick={onClick}
      className={`${buttonClasses} ${className}`}
    >
      {children}
    </button>
  );
};

export default Button;
