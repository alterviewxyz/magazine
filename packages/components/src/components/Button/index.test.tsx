import React from "react";
import { render, RenderResult } from "@testing-library/react";
import Button from "./";

describe("Button Component", () => {
  const getComponent = (): RenderResult => render(<Button> Hello </Button>);

  test("renders button text", () => {
    const { getByText } = getComponent();
    const text = getByText(/Hello/i);
    expect(text).toBeInTheDocument();
  });

  test("match snapshot 1", () => {
    const { container } = getComponent();
    expect(container.firstChild).toMatchSnapshot();
  });
});
