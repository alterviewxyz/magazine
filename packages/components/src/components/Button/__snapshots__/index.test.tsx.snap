// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component match snapshot 1 1`] = `
<button
  class="btn undefined"
>
   Hello 
</button>
`;
